tests/moe_ep/test_bf16_rank_major_cuda_backend_cpu.py

```
@@ -0,0 +1,308 @@
+    with pytest.raises(MoEEpConfigError, match=message):
+        backend.validate_forward(
+            _fake_inputs(tokens=tokens),
+            _exact_fleet(),
+            quantize_input=True,
+        )
+
+
+def test_forward_validation_rejects_dtype_drift():
+    backend = _backend()
+    with pytest.raises(
+        MoEEpConfigError, match="topk_ids must be torch.int32 or torch.int64"
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)